proposals/factory_constructors.md

```diff
@@ -139,6 +139,14 @@ public class IFactory
     public factory I(string param1) => ...;
     public static factory I(object param1) => ...;
 }
+class Widget
+{
+    static factory Widget() => new Widget(string.Empty, -1);
+    static factory Widget(Widget widget) => new Widget() { Title = widget.Title, Id = widget.Id };
```

Comment:
this seems problematic because you can't have member names with the name as hte containing type.  But maybe that can be relaxed here.

Reply:
That's why these would be emitted with unspeakable names. The only way to use them is like a constructor.